x509/pkix/pkix.go

```diff
@@ -15,6 +15,10 @@ import (
 	"github.com/zmap/zcrypto/encoding/asn1"
 )
 
+// LegacyNameString allows to specify legacy ZCrypto behaviour
+// for X509Name.String() in reverse order
+var LegacyNameString = false
```

Comment:
Thinking about this some more... what is your thought on how users of zcrypto should set `LegacyNameString`?  Since it is a global, maybe it is safer to not export it and have a Get/Set function for it that uses a RWMutex.  Although since it is global I guess it really only makes since to set it once during initialization.

Reply:
It does seem like an odd thing to change dynamically during execution. 

I think the idiomatic path here is to leave concurrency to the consumer. If they are going to do something that dynamically changes this across routines, the consumer should be responsible for the locking.

Reply:
in censys, we initialize it in the app once. there is no need for mutex or get/set.
either in `main()` or in `init() {
asn1.AllowPermissiveParsing = true
pkix.LegacyNameString = true
}`

Reply:
@dissoupov I just want to make sure we are not designing for a single downstream user's use case, since as an open-source project there may be many other users that pull in this functionality.  Drew's response makes sense though, so I am fine leaving it as-is.

Reply:
this is not intended to be changed at run time.
by default, there will be standard functionality as in Go x509.
for apps that need different behavior, it should be initialized only once.